src/main/java/Main.java

```diff
@@ -0,0 +1,111 @@
+import spark.ModelAndView;
+import spark.Session;
+import spark.Spark;
+import spark.template.mustache.MustacheTemplateEngine;
+
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Created by vasantia on 7/11/16.
+ */
+public class Main {
+    public static final String SESSION_USERNAME = "userName";
+    static Map<String, User> users = new HashMap<>();
+
+    public static void main(String[] args) {
+        Spark.init();
+
+        Spark.get("/", (request, response) -> {
+            HashMap m = new HashMap();
+            Session session = request.session();
+            String name = session.attribute(SESSION_USERNAME);
+            User user = users.get(name);
+
+            if(user == null){
+                return new ModelAndView(m, "index.html");
+            }
+            else {
+                m.put("user", user);
+                return new ModelAndView(m, "messages.html");
+            }
+        }, new MustacheTemplateEngine());
+
+        Spark.get("/create-user", ((request, response) -> {
+            HashMap m = new HashMap();
+            return new ModelAndView(m, "index.html");
+        }), new MustacheTemplateEngine());
+
+        Spark.post("/create-user", (request, response) -> {
+            String name = request.queryParams("submitName");
+            String password = request.queryParams("submitPassword");
+            User user = users.get(name);
+
+            if(user == null) {
+                user = new User(name, password);
+                users.put(name, user);
+                Session session = request.session();
```

Comment:
Given that a you use session a few times in this controller it would be better to move this line out of the if/else sections.